### PR TITLE
#172 Select checkboxes

### PR DESCRIPTION
**Pull request by NFSS10**
| - | - |
| --- | --- |
| Issue | https://github.com/NFSS10/peri-wysiwyg/pull/16 |
| Dependencies | -- |
| Decisions | • Created a select component that when opened shows a group of checkboxes <br>• Added tests:<br>&nbsp;&nbsp;&nbsp;&nbsp;• `should start with the correct state`<br>&nbsp;&nbsp;&nbsp;&nbsp;• `should change label via prop`<br>&nbsp;&nbsp;&nbsp;&nbsp;• `should change checked value in allowed checkboxes` |
| Animated GIF | select-checkboxes<br>imagem |

### CODE REVIEW

File: vue/components/ui/molecules/select-checkboxes/select-checkboxes.vue [hunk 1]
```diff
@@ -0,0 +1,95 @@
+<template>
+    <select-ripe
+        class="select-checkboxes"
+        v-bind:options="selectOptions"
+        v-bind:value="'checkbox-group'"
+        v-bind="{
+            autoScroll: false,
```
Comment (gcandal) [inline vue/components/ui/molecules/select-checkboxes/select-checkboxes.vue:7]: Is this really needed?

Comment (NFSS10) [reply vue/components/ui/molecules/select-checkboxes/select-checkboxes.vue:7]: Yes if we don't want to have it scroll to the bottom of the checkboxes-group

File: vue/test/components/select-checkboxes/select-checkboxes.test.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        assert.strictEqual(
+            component.find(".select > .select-container > .select-button").text(),
+            "Label Example"
+        );
+        const dropdown = component.find(".select > .select-container > .dropdown-container");
+        const checkboxes = dropdown.findAll(".checkboxes > .checkbox-group > .checkbox-item");
+        assert.strictEqual(checkboxes.length, 7);
+        assert.strictEqual(checkboxes.at(0).find(".checkbox.checked").exists(), false);
+        assert.strictEqual(
+            checkboxes.at(0).find(".checkbox > .checkbox-input > .label").text(),
```
Comment (gcandal) [inline vue/test/components/select-checkboxes/select-checkboxes.test.js:38]: Isn't `.at(0).text()` enough?

File: vue/test/components/select-checkboxes/select-checkboxes.test.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            "Bali"
+        );
+        assert.strictEqual(checkboxes.at(6).find(".checkbox.checked").exists(), false);
+        assert.strictEqual(checkboxes.at(6).find(".checkbox.error").exists(), true);
+        assert.strictEqual(
+            checkboxes.at(6).find(".checkbox > .checkbox-input > .label").text(),
+            "Tibet"
+        );
+
+        await checkboxes.at(0).find(".checkbox > .checkbox-input").trigger("click");
```
Comment (gcandal) [inline vue/test/components/select-checkboxes/select-checkboxes.test.js:157]: Can't you simply target `.checkbox`?

**Review by gcandal — CHANGES_REQUESTED** [review]
Looks good, let's just try to slightly simplify the tests.

**Review by gcandal — APPROVED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**